assets/js/modules/analytics/components/dashboard/DashboardUniqueVisitorsWidget.stories.js

```diff
@@ -0,0 +1,215 @@
+import DashboardUniqueVisitorsWidget from './DashboardUniqueVisitorsWidget';
+
+const areaName = 'moduleAnalyticsMain';
+const widgetSlug = 'analyticsUniqueVisitors';
+const currentEntityURL = 'https://www.example.com/example-page/';
+const options = {
+	startDate: '2020-08-11',
+	endDate: '2020-09-07',
+	dimensionFilters: { 'ga:channelGrouping': 'Organic Search' },
+	dimensions: [ 'ga:date', 'ga:channelGrouping' ],
+	metrics: [
+		{
+			expression: 'ga:users',
+			alias: 'Users',
+		},
+	],
+};
+const optionsEntityURL = {
+	...options,
+	url: currentEntityURL,
+};
+const optionsCompare = {
+	compareStartDate: '2020-07-14',
+	compareEndDate: '2020-08-10',
+	startDate: '2020-08-11',
+	endDate: '2020-09-07',
+	dimensionFilters: { 'ga:channelGrouping': 'Organic Search' },
+	dimensions: [ 'ga:channelGrouping' ],
+	metrics: [
+		{
+			expression: 'ga:users',
+			alias: 'Total Users',
+		},
+	],
+};
+const optionsCompareEntityURL = {
+	...optionsCompare,
+	url: currentEntityURL,
+};
+const WidgetAreaTemplate = ( args ) => {
+	return (
+		<WithRegistrySetup func={ args?.setupRegistry }>
+			<WidgetAreaRenderer slug={ areaName } />
+		</WithRegistrySetup>
+	);
+};
+
+export const Loaded = WidgetAreaTemplate.bind();
+Loaded.storyName = 'Loaded';
+Loaded.args = {
+	setupRegistry: ( { dispatch } ) => {
+		dispatch( STORE_NAME ).receiveGetReport( getAnalyticsMockResponse( optionsCompare ), { options: optionsCompare } );
+		dispatch( STORE_NAME ).receiveGetReport( getAnalyticsMockResponse( options ), { options } );
+	},
+};
+
+export const Loading = WidgetAreaTemplate.bind();
+Loading.storyName = 'Loading';
+Loading.args = {
+	setupRegistry: ( { dispatch } ) => {
+		dispatch( STORE_NAME ).receiveGetReport( getAnalyticsMockResponse( optionsCompare ), { options: optionsCompare } );
+		dispatch( STORE_NAME ).receiveGetReport( getAnalyticsMockResponse( options ), { options } );
+		dispatch( STORE_NAME ).startResolution( 'getReport', [ options ] );
+		dispatch( STORE_NAME ).startResolution( 'getReport', [ optionsCompare ] );
+	},
+};
+
+export const DataUnavailable = WidgetAreaTemplate.bind();
+DataUnavailable.storyName = 'Data Unavailable';
+DataUnavailable.args = {
+	setupRegistry: ( { dispatch } ) => {
+		dispatch( STORE_NAME ).receiveGetReport( [], { options } );
+	},
+};
+
+export const Error = WidgetAreaTemplate.bind();
+Error.storyName = 'Error';
+Error.args = {
+	setupRegistry: ( { dispatch } ) => {
+		const error = {
+			code: 'missing_required_param',
+			message: 'Request parameter is empty: metrics.',
+			data: {},
+		};
+		dispatch( STORE_NAME ).receiveError( error, 'getReport', [ options ] );
+		dispatch( STORE_NAME ).finishResolution( 'getReport', [ options ] );
+	},
+};
+
+export const LoadedEntityURL = WidgetAreaTemplate.bind();
+LoadedEntityURL.storyName = 'Loaded with entity URL set';
+LoadedEntityURL.args = {
+	setupRegistry: ( registry ) => {
+		const { dispatch } = registry;
```

Comment:
The destructuring makes it a bit cleaner when the full `registry` isn't needed, but when it is it's probably better to call `registry.dispatch` below and modify the formatting of those lines to be spread across multiple as they're already a bit too long as-is.